src/core/cookie_settings/CookieManagementService.ts

```diff
@@ -0,0 +1,105 @@
+import { permissionStatuses, permissionStrategy } from './constants';
+import { Permission } from './Permission';
+import { PermissionStatusWatcher } from './PermissionStatusWatcher';
+import { PermissionStorage, StrategyStorage } from './Storage';
+import type { CookiePermissionResolveStrategy, CookiePermissionStatus } from './types';
+
+type SettingsListener = (p: Map<string, Permission>) => void;
+
+export class CookieManagementService {
+  private permissions = new Map<string, PermissionStatusWatcher>();
+  private permissionStorage = new PermissionStorage();
+  private strategyStorage = new StrategyStorage();
+  private currentStrategy: CookiePermissionResolveStrategy = permissionStrategy.prompt;
+
+  static statusFromStrategy(
+    strategy: CookiePermissionResolveStrategy,
+  ): CookiePermissionStatus {
+    switch (strategy) {
+      case permissionStrategy.applyAll:
+        return permissionStatuses.granted;
+
+      case permissionStrategy.rejectAll:
+        return permissionStatuses.denied;
+
+      case permissionStrategy.prompt:
+        return permissionStatuses.promptNeeded;
+    }
+  }
+
+  /**
+   * Strategy it kind of "fallback behavior"
+   * For example - when user choose "Apply all"
+   * This service will allow to use any cookies for feature requests
```

Comment:
@katluk is that permitted?

Reply:
Yes, for now it is ok. During the next stage we are going to introduce the mechanism to manage cookies.